src/terms/ewald.jl

```diff
@@ -89,6 +76,11 @@ function energy_ewald(lattice, recip_lattice, charges, positions; η=nothing, fo
     poslims = [maximum(rj[i] - rk[i] for rj in positions for rk in positions) for i in 1:3]
     Rlims = estimate_integer_lattice_bounds(lattice, max_erfc_arg / η, poslims)
 
+    # Check if some coordinates are not used.
+    is_dim_trivial = iszero.(eachcol(lattice))
+    max_shell(n, trivial) = trivial ? 0 : n
+    Rlims = max_shell.(Rlims, is_dim_trivial)
```

Comment:
Indeed and for Coulomb the situation is more tricky. E.g. if you want true 2D Coulomb you need to use a `ln(r)` instead of a `1/r` potential. On the other hand if you want do to a 3D Coulomb on objects constrained onto a 1D/2D system with infinite separation in between what you do seems reasonable, but it's a pretty specific use case.

My tendency is therefore to not allow `Ewald` as we implement it for non-3D and for the user to do something manual using the pairwise potential.

Reply:
I added a sentence.